### ISSUE TITLE
[DragonflyBSD] jconstruct-tui hangs on Applying skel dir

### ISSUE THREAD

**Issue by BigB84** [cbsd/cbsd#710]
Mandatory info for bug reports:

DragonflyBSD version ( uname -a ): 
`6.2-RELEASE DragonFly v6.2.2-RELEASE`

CBSD version ( cbsd version ): 
`13.1.1`

Steps:

- Installed dfly using installer from official website.
- Installed cbsd using `pkg install cbsd`
- Completed `cbsd init`
- Enabled pf and basic ruleset allowing traffic on loopback, in, out and ssh
- Reboot, cbsd daemon enabled. Lets create some jails.
- **FIRST GLITCH:** cbsd jconstruct-tui couldn't find bsdconfig so I blindly copied it from this repo: https://github.com/cbsd/cbsd_useful_stuff
- Now cbsd jconstruct-tui could launch
- Using default template I try to create jail
- I have no bases yet, so it needs to be downloaded/built/extracted first.
- **SECOND GLITCH:** Build base crashes with error:

```bash
 error: pathspec 'DragonFly_RELEASE_6.2' did not match any file(s) known to git
 + _ret=1
 + set +o xtrace
 cd /usr/jails/src/src_6.2/src && /usr/local/bin/git rev-parse --short DragonFly_RELEASE_6.2
 fatal: your current branch 'master' does not have any commits yet
```

- **THIRD GLITCH:** I tried again but different methods but both extract and download hangs on:

```bash
 Please wait: this will take a while...
 hammer2: pfs_create(/usr/jails/jails-data/jail1-data): SUCCESS
 Applying skel dir template from: /usr/jails/share/DragonFly-jail-skel
```

It won't pass further. I was waiting long, then cancelled by CTRL+C.
I checked htop while it was doing but processes weren't explicitly consuming CPU or Memory.
I tried debugging cbsd with setting up env CBSD_DEBUG=1 but it is so verbose that I gave up.

**Comment by olevole** [comment]
**> FIRST GLITCH: cbsd jconstruct-tui couldn't find bsdconfig so I blindly copied it from this repo: https://github.com/cbsd/cbsd_useful_stuff**

Yes,  this is described in https://github.com/cbsd/cbsd/blob/develop/README.DragonflyBSD.md but it may be necessary to display a separate message in post-message for DragonflyBSD platform only ( or add into DFLY-ports ). We have to think about it ;-)


**> THIRD GLITCH: I tried again but different methods but both extract and download hangs on:**

It seems this is due to the fact that the dragonfly has changed the architecture format (and target_arch - `uname -p` ):

```
uname -m
x86_64
```
to fix: please apply latest changes from main git:

```
fetch -o /usr/local/cbsd/etc/defaults/DragonFly-bases.conf https://raw.githubusercontent.com/cbsd/cbsd/develop/etc/defaults/DragonFly-bases.conf
fetch -o /usr/local/cbsd/subr/build.subr  https://raw.githubusercontent.com/cbsd/cbsd/develop/subr/build.subr
fetch -o /usr/local/cbsd/tools/repo https://raw.githubusercontent.com/cbsd/cbsd/develop/tools/repo
chmod +x /usr/local/cbsd/tools/repo
cbsd initenv
```

Now you will be able to fetch base & create a container via **jconstruct-tui** or **CBSDfile** or **jcreate**:
```
cbsd jcreate jname=test pkg_bootstrap=0
cbsd jcreate jname=test1 pkg_bootstrap=0 runasap=1
```

On other questions, I will take some timeout.

By the way, you are not using the latest version of CBSD. 

@tuxillo maybe you know why DragonFlyBSD doesn't build the latest version of CBSD from FreeBSD ports?  https://cgit.freebsd.org/ports/tree/sysutils/cbsd  << the last few CBSD versions, including the latest one, are builds on a dragonfly without any modifications.

**Comment by olevole** [comment]
upd: **THIRD GLITCH** fixed. To apply - fetch changes from main git:

```
fetch -o /usr/local/cbsd/etc/defaults/DragonFly-bases.conf https://raw.githubusercontent.com/cbsd/cbsd/develop/etc/defaults/DragonFly-bases.conf

fetch -o /usr/local/cbsd/subr/build.subr  https://raw.githubusercontent.com/cbsd/cbsd/develop/subr/build.subr

fetch -o /usr/local/cbsd/tools/repo https://raw.githubusercontent.com/cbsd/cbsd/develop/tools/repo

chmod +x /usr/local/cbsd/tools/repo
cbsd initenv
```

**Comment by tuxillo** [comment]
> @tuxillo maybe you know why DragonFlyBSD doesn't build the latest version of CBSD from FreeBSD ports? https://cgit.freebsd.org/ports/tree/sysutils/cbsd << the last few CBSD versions, including the latest one, are builds on a dragonfly without any modifications.

I'll check and let you know.

**Comment by tuxillo** [comment]
Next week the newer version will be available in DPorts.

**Comment by olevole** [comment]
**SECOND GLITCH** fixes: https://github.com/cbsd/cbsd/commit/ca4b36d3d1b28f4907257f2c40d205a49a63446f

to fix: please apply latest changes from main git:
```
fetch -o /usr/local/cbsd/share/DragonFly-git_branches.conf https://raw.githubusercontent.com/cbsd/cbsd/develop/share/DragonFly-git_branches.conf
fetch -o /usr/local/cbsd/system/buildworld https://raw.githubusercontent.com/cbsd/cbsd/develop/system/buildworld
fetch -o /usr/local/cbsd/system/installworld https://raw.githubusercontent.com/cbsd/cbsd/develop/system/installworld
chmod +x /usr/local/cbsd/system/buildworld /usr/local/cbsd/system/installworld
```

this fixes the commands to get the container base: `cbsd srcup` , `cbsd world`

fixes will be included in the next release - **13.1.15**.
Please re-open issue if problem remain  after applying the patches.

Thanks.
